documentation/tests/integration/components/list-item-test.js

```diff
@@ -0,0 +1,151 @@
+    assert.dom('#test-list-item').hasAttribute('href', '/dummy');
+  });
+
+  test('it should render a <button> if @onClick is passed', async function (assert) {
+    this.set('onClick', () => {});
+    await render(hbs`
+      <Cut::ListItem id="test-list-item" @onClick={{this.onClick}} />
+    `);
+    assert.dom('#test-list-item').hasTagName('button');
+  });
+
+  test('it should render a <a> link with custom "target" and "rel" attributes if they are passed as attributes', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" @href="#" target="test-target" rel="test-rel" />`
+    );
+    assert.dom('#test-list-item').hasAttribute('target', 'test-target');
+    assert.dom('#test-list-item').hasAttribute('rel', 'test-rel');
+  });
+
+  test('it should render a <a> link withhout "target" and "rel" attributes if @isHrefExternal is false', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" @href="#" @isHrefExternal={{false}} />`
+    );
+    assert.dom('#test-list-item').doesNotHaveAttribute('target');
+    assert.dom('#test-list-item').doesNotHaveAttribute('rel');
+  });
+
+  test('it should spread all the attributes passed to the <div> element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('div#test-list-item').hasClass('my-class');
+    assert.dom('div#test-list-item').hasAttribute('data-test1');
+    assert.dom('div#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should spread all the attributes passed to the <a> element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" @href="#" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('a#test-list-item').hasClass('my-class');
+    assert.dom('a#test-list-item').hasAttribute('data-test1');
+    assert.dom('a#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should spread all the attributes passed to the <LinkTo> element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem @route="index" id="test-list-item" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('a#test-list-item').hasClass('my-class');
+    assert.dom('a#test-list-item').hasAttribute('data-test1');
+    assert.dom('a#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should spread all the attributes passed to the <button> element', async function (assert) {
+    this.set('onClick', () => {});
+
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" @onClick={{this.onClick}} class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('button#test-list-item').hasClass('my-class');
+    assert.dom('button#test-list-item').hasAttribute('data-test1');
+    assert.dom('button#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
```

Comment:
I don't think you need to test the spreading for all the variants. since you're spreading on the `Hds::Interactive`, if it works for one variant it works for all of them (testing that all the variants are spreading is _our_ (HDS) concern, and is already covered by our tests, so you should consider it a "safe" assumption)

```suggestion
```